### PR TITLE
feat(service-portal): UI fixes on access control 

### PR DESCRIPTION
**Pull request by disaerna**
# UI fixes on access control flow

## What

* Fix DatePicker to be able to be positioned on bottom of page, if so then render options above input.
* Refactor modal into seperate component
* Refactor translations 
* Add "extract strings" command
* Refactor folder structure to simplify extraxt strings command
* Add "onSave Modal"
* List accesses on modals
* Add default date to 3 months
* Remove table structure and add columns instead (scaling on mobile screens were off)
* Replace kennitala placeholder with numbers

## Why

New design and requests from Hrefna

## Screenshots / Gifs

Screenshot 2022-03-21 at 14 01 41

Screenshot 2022-03-21 at 21 12 11

Screenshot 2022-03-21 at 14 02 11

Screenshot 2022-03-21 at 14 02 04

Screenshot 2022-03-21 at 21 00 37

Screenshot 2022-03-21 at 21 00 48

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/service-portal/settings/access-control/src/screens/Access/Access.tsx [hunk 1]
```diff
@@ -209,153 +212,158 @@ function Access() {
...
-                            })}
-                          </Button>
-                        </Box>
-                      </Stack>
-                    </Box>
+                      //
+                      text={
+                        `${authDelegation?.to?.name} mun missa umboð fyrir eftirfarandi:` ||
+                        ''
+                      }
```
Comment (eirikurn) [inline libs/service-portal/settings/access-control/src/screens/Access/Access.tsx:268]: Comment? formatMessage? `|| ''`? :)

File: libs/service-portal/settings/access-control/src/screens/Access/Access.tsx [hunk 2]
```diff
@@ -209,153 +212,158 @@ function Access() {
...
+                  id="access-save-modal"
+                  title={formatMessage({
+                    id: 'sp.settings-access-control:access-save-modal-content',
+                    defaultMessage: 'Ertu viss um að þú viljir veita umboðið?',
+                  })}
+                  //
+                  text={
+                    `${authDelegation?.to?.name}  mun fá umboð fyrir eftirfarandi:` ||
+                    ''
+                  }
```
Comment (eirikurn) [inline libs/service-portal/settings/access-control/src/screens/Access/Access.tsx:310]: Same

File: libs/service-portal/settings/access-control/src/screens/Access/Access.tsx [hunk 3]
```diff
@@ -209,153 +212,158 @@ function Access() {
...
             </Inline>
           </Box>
 
           <div className={styles.datePickerFix}>
-            <T.Table>
-              <T.Head>
-                <T.Row>
-                  <T.HeadData>
-                    <Text variant="small" color="blue600">
+            <Box>
```
Comment (eirikurn) [inline libs/service-portal/settings/access-control/src/screens/Access/Access.tsx:329]: Nice! Can we now get rid of `<div className={styles.datePickerFix}>`?

File: libs/service-portal/settings/access-control/src/screens/Access/Access.tsx [hunk 4]
```diff
@@ -209,153 +212,158 @@ function Access() {
...
+                    icon="close"
+                    onClick={() => setCloseModalOpen(true)}
+                  >
+                    {formatMessage({
+                      id: 'sp.settings-access-control:access-remove-delegation',
+                      defaultMessage: 'Eyða umboði',
+                    })}
+                  </Button>
+                  {closeModalOpen && (
+                    <AccessModal
```
Comment (eirikurn) [inline libs/service-portal/settings/access-control/src/screens/Access/Access.tsx:256]: Is this the accessibly correct way to open and close modals? Does it handle focus correctly? E.g. when going in and out of the modal?

File: libs/service-portal/settings/access-control/src/components/AccessItem/AccessItem.tsx [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+                      placeholder={undefined}
+                      onChange={(value) => onChange(item, value)}
+                    />
+                  </div>
+                </Box>
+              </GridColumn>
+            </GridRow>
+            <Box paddingY={1}>
+              <Divider />
+            </Box>
```
Comment (eirikurn) [inline libs/service-portal/settings/access-control/src/components/AccessItem/AccessItem.tsx:191]: IMHO the vertical spacing on mobile is a bit too tight around the divider and a bit too airy around the description. What do you think?

File: libs/service-portal/settings/access-control/src/components/AccessItem/AccessItem.tsx [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+                      defaultValue={
+                        existingScope?.name
+                          ? existingScope.validTo
+                          : format(defaultDate, 'yyyy-MM-dd')
+                      }
+                      locale={lang}
+                      placeholder={undefined}
+                      onChange={(value) => onChange(item, value)}
+                    />
+                  </div>
```
Comment (eirikurn) [inline libs/service-portal/settings/access-control/src/components/AccessItem/AccessItem.tsx]: The list takes a lot of space at rest on mobile with all the date pickers showing.

Is it weird to hide the datepicker on mobile and expand the row when the row is checked?

File: libs/service-portal/settings/access-control/src/screens/Access/Access.tsx [hunk 7]
```diff
@@ -209,153 +212,158 @@ function Access() {
...
                       })}
                     </Text>
-                  </T.HeadData>
-                  <T.HeadData>
-                    <Text variant="small" color="blue600">
+                  </GridColumn>
+                  <GridColumn
+                    span={['12/12', '12/12', '4/12', '5/12']}
+                    className={styles.column}
+                  >
```
Comment (eirikurn) [inline libs/service-portal/settings/access-control/src/screens/Access/Access.tsx:348]: Can we make the column titles better align horizontally with each cell?

File: libs/service-portal/settings/access-control/src/components/AccessItem/AccessItem.tsx [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+          <>
+            <GridRow key={index}>
+              <GridColumn
+                span={['12/12', '12/12', '3/12']}
+                className={styles.item}
+              >
+                <Box
+                  paddingBottom={isLastItem ? 'p3' : 'p1'}
+                  paddingTop={isFirstItem ? 'p3' : 'p1'}
+                  paddingLeft={isFirstItem ? 0 : [0, 0, 4]}
```
[the remark this answers is not in the document]
Comment (eirikurn) [reply libs/service-portal/settings/access-control/src/components/AccessItem/AccessItem.tsx:125]: Is this better? I don't know.

image

[the remark this answers is not in the document]
Comment (eirikurn) [reply libs/service-portal/settings/access-control/src/components/AccessItem/AccessItem.tsx:125]: Fyi, I would really like to explore different UX to configure the validity of the scopes ... maybe in a follow up :)

**Review by eirikurn — CHANGES_REQUESTED** [review]
So happy to see the datepicker overflow issue solved and better mobile design. Great improvement!

Most of my comments are minor, so let me know if you want to skip some of them for now.

**Review by Arnarkari93 — APPROVED** [review]
LGTM from the `island-ui` perspective.

File: libs/service-portal/settings/access-control/src/screens/Access/Access.tsx (hunk 1 above)
Comment (disaerna) [reply libs/service-portal/settings/access-control/src/screens/Access/Access.tsx]: woops - solved

File: libs/service-portal/settings/access-control/src/screens/Access/Access.tsx (hunk 2 above)
Comment (disaerna) [reply libs/service-portal/settings/access-control/src/screens/Access/Access.tsx]: ✅

File: libs/service-portal/settings/access-control/src/screens/Access/Access.tsx (hunk 3 above)
Comment (disaerna) [reply libs/service-portal/settings/access-control/src/screens/Access/Access.tsx:359]: Yes def! ✅

File: libs/service-portal/settings/access-control/src/screens/Access/Access.tsx (hunk 4 above)
Comment (disaerna) [reply libs/service-portal/settings/access-control/src/screens/Access/Access.tsx]: Oh lord - it was not. Thank you so much 🙏  I think that I fixed it ? 🤔

File: libs/service-portal/settings/access-control/src/components/AccessItem/AccessItem.tsx (hunk 5 above)
Comment (disaerna) [reply libs/service-portal/settings/access-control/src/components/AccessItem/AccessItem.tsx]: Agree - fixed ✅

File: libs/service-portal/settings/access-control/src/components/AccessItem/AccessItem.tsx (hunk 6 above)
Comment (disaerna) [reply libs/service-portal/settings/access-control/src/components/AccessItem/AccessItem.tsx]: Yeb agree - ✅

File: libs/service-portal/settings/access-control/src/components/AccessItem/AccessItem.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (disaerna) [reply libs/service-portal/settings/access-control/src/components/AccessItem/AccessItem.tsx]: Indent fixed ✅  
UX stuff - yes, the new design is on Figma, I'll send you a link 👍

File: libs/service-portal/settings/access-control/src/components/AccessItem/AccessItem.tsx [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+                  paddingLeft={[0, 0, 3]}
+                >
+                  <div className={cs(isSelected ? undefined : styles.hidden)}>
+                    <DatePickerController
+                      id={`${item.model}.validTo`}
+                      size="sm"
+                      label={formatMessage({
+                        id:
+                          'sp.settings-access-control:access-item-datepicker-label',
+                        defaultMessage: 'Dagsetning til',
```
[the remark this answers is not in the document]
Comment (disaerna) [reply libs/service-portal/settings/access-control/src/components/AccessItem/AccessItem.tsx]: Yeb, absolute ✅  fixed

File: libs/service-portal/settings/access-control/src/screens/Access/Access.tsx (hunk 7 above)
Comment (disaerna) [reply libs/service-portal/settings/access-control/src/screens/Access/Access.tsx]: wow - yes ✅ fixed

File: libs/service-portal/settings/access-control/src/components/AccessItem/AccessItem.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (disaerna) [reply libs/service-portal/settings/access-control/src/components/AccessItem/AccessItem.tsx]: BTW - I updated the screenshots 😸

**Review by eirikurn — APPROVED** [review]
Great job!

**Pull request merged**